forms/keyword/file_search_form.go

```diff
@@ -0,0 +1,120 @@
+package forms
+
+import (
+	"mime/multipart"
+
+	. "go-crawler-challenge/forms"
+	"go-crawler-challenge/helpers"
+	"go-crawler-challenge/models"
+
+	"github.com/beego/beego/v2/client/orm"
+	"github.com/beego/beego/v2/core/logs"
+	"github.com/beego/beego/v2/core/validation"
+)
+
+type FileSearchForm struct {
+	File       multipart.File
+	FileHeader *multipart.FileHeader
+	User       *models.User `valid:"Required"`
+
+	keywordList []string
+}
+
+func (form *FileSearchForm) Valid(validation *validation.Validation) {
+	if form.File == nil {
+		err := validation.SetError("File", ValidationMessages["RequireFile"])
+		if err == nil {
+			logs.Warning("Set validation error failed")
+		}
+
+		return
+	}
+
+	if !helpers.CheckMatchFileType(form.FileHeader, []string{KeywordUploadContentTypeCSV}) {
+		err := validation.SetError("File", ValidationMessages["InvalidFileType"])
+		if err == nil {
+			logs.Warning("Set validation error failed")
+		}
```

Comment:
>	if !valid {
		return validator.Errors[0]
	}
	
As we showing only first error, shall we return early if the any validation fails?

```Go
        .....
        .....
	if err == nil {
		logs.Warning("Set validation error failed")
	}
	
	return
```